### PR TITLE
feat(EMS-2370): No PDF - Your buyer - Trading history page

### PR DESCRIPTION
**Pull request by Zainzzkk**
## Introduction :pencil2:
This PR adds the trading history page in your buyer section

## Resolution :heavy_check_mark:
- Added trading history to your buyer section
- Added tests and e2e tests

## Miscellaneous :heavy_plus_sign:
- Added DATA-CY variable for HINT to core-page-variables

### CODE REVIEW

File: e2e-tests/insurance/cypress/e2e/journeys/your-buyer/trading-history/trading-history-page.spec.js [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+  before(() => {
+    cy.completeSignInAndGoToApplication({}).then(({ referenceNumber: refNumber }) => {
+      referenceNumber = refNumber;
+
+      cy.startInsuranceYourBuyerSection({});
+
+      // cy.completeAndSubmitCompanyOrOrganisationForm({});
+      // cy.completeAndSubmitConnectionToTheBuyerForm({});
+
+      url = `${Cypress.config('baseUrl')}${INSURANCE_ROOT}/${referenceNumber}${TRADING_HISTORY}`;
```
Comment (ttbarnes) [inline e2e-tests/insurance/cypress/e2e/journeys/your-buyer/trading-history/trading-history-page.spec.js:38]: Can we have a `baseUrl` const just before the `context`?

File: e2e-tests/constants/field-ids/insurance/your-buyer/index.js [hunk 2]
```diff
@@ -14,4 +14,6 @@ export const YOUR_BUYER = {
   CONNECTION_WITH_BUYER: 'exporterIsConnectedWithBuyer',
   CONNECTION_WITH_BUYER_DESCRIPTION: 'connectionWithBuyerDescription',
   TRADED_WITH_BUYER: 'exporterHasTradedWithBuyer',
+  OUTSTANDING_PAYMENTS: 'outstandingPayments',
+  FAILED_PAYMENTS: 'failedPayments',
```
[the remark this answers is not in the document]
Comment (ttbarnes) [reply e2e-tests/constants/field-ids/insurance/your-buyer/index.js:18]: Ignore, just realised they are yes/no

File: e2e-tests/insurance/cypress/e2e/journeys/your-buyer/trading-history/trading-history-page.spec.js [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+
+  //     describe('when going back to the page', () => {
+  //       it('should have the submitted values', () => {
+  //         cy.navigateToUrl(url);
+
+  //         noRadioInput().should('be.checked');
+  //       });
+  //     });
+  //   });
+  // });
```
Comment (ttbarnes) [inline e2e-tests/insurance/cypress/e2e/journeys/your-buyer/trading-history/trading-history-page.spec.js:181]: Why the commented code 😛

File: e2e-tests/pages/shared/index.js [hunk 4]
```diff
@@ -14,6 +14,7 @@ const body = () => cy.get('[data-cy="body"]');
 const form = () => cy.get('[data-cy="form"]');
 const heading = () => cy.get('[data-cy="heading"]');
 const headingCaption = () => cy.get('[data-cy="heading-caption"]');
+const hint = () => cy.get('[data-cy="hint"]');
```
Comment (ttbarnes) [inline e2e-tests/pages/shared/index.js:17]: This is already available in `field`.

File: src/ui/templates/insurance/your-buyer/trading-history.njk [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+        <span class="govuk-caption-xl" data-cy="heading-caption">
+          {{ CONTENT_STRINGS.HEADING_CAPTION }}
+        </span>
+
+        <h1 class="govuk-heading-xl" data-cy="{{ DATA_CY.HEADING }}">{{ CONTENT_STRINGS.PAGE_TITLE }}</h1>
+
+        {{ govukHint({
+          text: CONTENT_STRINGS.HINT,
+          attributes: {
+          "data-cy": DATA_CY.HINT
```
Comment (ttbarnes) [inline src/ui/templates/insurance/your-buyer/trading-history.njk:50]: Minor - indentation

File: e2e-tests/insurance/cypress/e2e/journeys/your-buyer/trading-history/trading-history-page.spec.js (hunk 3 above)
Comment (Zainzzkk) [reply e2e-tests/insurance/cypress/e2e/journeys/your-buyer/trading-history/trading-history-page.spec.js:181]: to be tested later once I can renable

File: e2e-tests/pages/shared/index.js (hunk 4 above)
Comment (Zainzzkk) [reply e2e-tests/pages/shared/index.js:17]: that one has a field id before it and this hint is just data-cy hint

Comment (ttbarnes) [reply e2e-tests/pages/shared/index.js:17]: Can we have the same approach as `CONNECTION_WITH_BUYER`? That has a hint displayed before the radio.

It's strange to have a `hint` like this not related to a field.

**Review by ttbarnes — APPROVED** [review]

**Pull request merged**